lib/groundlight/src/groundlight.cpp

```diff
@@ -381,6 +380,13 @@ detector_list get_detector_list(const char *endpoint, const char *apiToken) {
     strcpy(_detector_list[i].name, detectors[i]["name"]);
     strcpy(_detector_list[i].query, detectors[i]["query"]);
     strcpy(_detector_list[i].group_name, detectors[i]["group_name"]);
+    if (!detectors[i]["metadata"].isNull()) {
+          String metadataStr;
+          serializeJson(detectors[i]["metadata"], metadataStr);
+          strlcpy(_detector_list[i].metadata, metadataStr.c_str(), sizeof(_detector_list[i].metadata));
+      } else {
+          _detector_list[i].metadata[0] = '\0'; 
+      }
```

Comment:
An awesome bonus here would be to check the length of the string before you copy it into the char array. We've had memory overflow problems before on, for example, detectors with ridiculously long queries.

Reply:
I thought if I use strlcpy to copy the string, it specifies the size of string so that overflow won't happen?

Reply:
Admittedly not strictly a part of your PR, but just above here are some unsafe strcpy calls